### DISCUSSION TITLE
Pitote tube Q

### DISCUSSION THREAD

**Discussion by stefan63** [DIY-Flow-Bench/DIY-Flow-Bench#247] [General]
Since we already have a ref atmospheric from the BME do is a differntial pitot tube realy be necessary, can´t we simply use a single bent tube (with a dressed tube end) ?

**Comment by DeeEmm** [comment]
I did have this in the code originally (it is still there but commented out). The comparison is made between the pitot tube as the reference pressure (total pressure) and pRef as the differential (static pressure), But this does have some inaccuracy as the static pressure should ideally be taken from the same point as the pitot tube is measuring (within the port). This is easy with a differential style pressure sensor and a proper pitot probe, no so much with a single ended sensor and / or tube

sensors.cpp
```
// Convert Pitot pressure to velocity (m3/min)
// airVelocity = sqrt(2 * (pitotPressure - sensorVal.PRefKPA) / airDensity );
airVelocity = sqrt(2 * (pitotPressure) / airDensity );
```

However even with the single ended sensor it will work, just unsure on how much of an accuracy hit it will cause as it will throw the velocity calc out. (previously with only pressure we needed to interpret the results)

(NOTE: The delta could also be made between Pitot and Diff pressure, as this tap point would be above the orifice in a differential style bench. )

I guess for the purposes of identifying fast / slow areas of the port the actual unit of measurement in unimportant, we just need to identify greater than or less than, which just requires a point of reference. Even using Baro as the static pressure would do that, but the resultant velocity value would be way out of course. 

As I often say with instrumentation - it can happily measure in 'potatoes' as long as we can interpret the results. Accurate units of measurement are only important when we need to compare or measure against a standard. 

I had a friend who was absolutely obsessed with making sure that the temperature gauge in his hot rod accurately displayed the actual temperature in deg C. He completely missed the point that the data trend was far more important than the actual number. 

There are plans on the roadmap to implement a simple manometer or coloured display to give a quick and easy visual representation. When you consider that we can zero out the delta and so are easily able to determine greater than / less than. I guess if we could figure out what the offset was we could even add an offset to he velocity value to bring it more in line with what a proper pitot probe would measure.

There is definitely value in expanding the pitot functionality and adding functionality to the configuration:

- Ability to add an offset to velocity calc to allow for single ended probe (prob also good idea to tune differential probe as well)
- Ability to select static pressure source (Baro / pRef / pDiff / Pitot)
- Ability to change measurements

I will add these to the roadmap for V3

**Reply by stefan63** [reply]
Nicely explained as usual :)

**Comment by DeeEmm** [comment]
Created #249 - Pitot Improvements

**Comment by KirikauKiwi** [comment]
I think it would almost be impossible to get a useable (not error riddled) pitot measurement using a single port "pitot" tube in the tight confines of a cylinder head port, you may get away with this possibly on the outside of a car if both the measurement ports were in a similar large area in this case the flow disturbances between to the two measurement points could be similar / minimal.

Trying to measure P_static at atmospheric pressure outside of a cylinder heads port and measuring P_total inside the port where the localised flow disturbances inside the port are definitely not the same as the flow disturbances at the P-static location, doing this would introduce errors in calculation. (guessing they would be large).

A single port probe could be used to measure P_total and make comparisons to the changes in P_total from various locations in the port.

You could use the PRef as P_total and PDiff  as P_static to calculate ∆P, this would require a manual reconfiguration of the DIYFB shield or you could use Pref or PDiff as a differential sensor for Pitot measurements again most likely to require a manual reconfiguration.

Depending on the type of flow bench being used the end user could build their own version of the shield which may only need one or two of the pressure sensors if they want to reduce costs of the build and manually reconfigure their setup if / when they want to conduct port pressure change/ velocity measurements.

**Comment by KirikauKiwi** [comment]
@stefan63 about your comment above

> can´t we simply use a single bent tube (with a dressed tube end) ? 

Is this because the cost of but a U-bent Pitot probe and / or the difficulties in making one in the conventional way tube in a tube.

I am currently working on the 3D design of a Pitot probe set for inlet and exhaust flows, that will be printable with the likes of SLA (resin) and SLS, SLM (powder bed fusion) 3D printers. The designs will be available as part of the DIYFB project, once I have conducted a test print etc.

**Reply by stefan63** [reply]
Not a problem building one just that a single tube will become more compact in design when you intend to use it for hand probing.

**Reply by stefan63** [reply]
This might be of interest in regards of pitot based benches: 
https://www.researchgate.net/figure/General-view-of-all-3D-printed-parts-The-first-design-of-a-Pitot-tube-probe-A-The_fig1_319991388
